lib/cocoapods/validator.rb

```diff
@@ -72,7 +72,7 @@ def validate
       a_spec = spec
       if spec && @only_subspec
         subspec_name = @only_subspec.start_with?(spec.root.name) ? @only_subspec : "#{spec.root.name}/#{@only_subspec}"
-        a_spec = spec.subspec_by_name(subspec_name)
+        a_spec = spec.subspec_by_name(subspec_name, true, true)
```

Comment:
This will scan for all subspecs including test specs. If we find the requested spec is a test_spec we bail out below.